seed_data/management/commands/alter_data.py

```diff
@@ -178,26 +185,43 @@ def clear_future_runs(user=None, course=None):  # pylint: disable=unused-argumen
 @accepts_or_calculates_now
 def add_past_failed_run(user=None, course=None, now=None, grade=DEFAULT_FAILED_GRADE, add_if_exists=False):
     """Adds a past failed course run for a given user and course"""
+    chosen_past_run = get_past_course_run(user, course, now, add_if_exists)
+
+    # Add enrollment and failed grade for course run
+    CachedEnrollmentHandler(user).set_or_create(course_run=chosen_past_run)
+    CachedCurrentGradeHandler(user).set_or_create(course_run=chosen_past_run, grade=grade)
+    return chosen_past_run
+
+
+def get_past_course_run(user=None, course=None, now=None, add_if_exists=False):
+    """Loop through past course runs and find one without CachedCurrentGrade data"""
     chosen_past_run = None
     past_runs = CourseRun.objects.filter(
         course=course,
         end_date__lt=now,
     ).exclude(end_date=None).order_by('-end_date').all()
-    # Loop through past course runs and find one without CachedCurrentGrade data
     for past_run in past_runs:
         if not CachedCurrentGradeHandler(user).exists(past_run):
             chosen_past_run = past_run
             continue
         elif not add_if_exists:
-            raise Exception("Past failed course run already exists (id: {}, title: {})".format(
+            raise Exception("Past failed/passed course run already exists (id: {}, title: {})".format(
                 past_run.id,
                 past_run.title
             ))
     if not chosen_past_run:
         raise Exception("Can't find past run w/o CachedCurrentGrade")
-    # Add enrollment and failed grade for course run
+    return chosen_past_run
+
+
+@accepts_or_calculates_now
+def add_past_passed_run(user=None, course=None, now=None, grade=DEFAULT_GRADE, add_if_exists=False):
```

Comment:
tiny nitpick: both `add_past_failed_run` and `add_past_passed_run` depend on `get_past_course_run`. i think it makes sense to move the definition of `get_past_course_run` above the definition of `add_past_failed_run`